src/flexmeasures_client/client.py

```diff
@@ -0,0 +1,287 @@
+        """Post sensor data for the given time range."""
+        json = dict(
+            sensor=f"{entity_address}.{sensor_id}",
+            start=pd.Timestamp(
+                start
+            ).isoformat(),  # for example: 2021-10-13T00:00+02:00
+            duration=pd.Timedelta(duration).isoformat(),  # for example: PT1H
+            values=values,
+            unit=unit,
+        )
+        if prior:
+            json["prior"] = prior
+
+        response, status = await self.request(
+            uri="sensors/data",
+            json=json,
+        )
+        check_for_status(status, 200)
+        print("Sensor data sent successfully.")
+
+    async def post_schedule_trigger(
+        self,
+        sensor_id: int,
+        start: str,
+        soc_unit: str,
+        soc_at_start: float,
+        soc_targets: list,
+        consumption_price_sensor: int | None = None,
+        production_price_sensor: int | None = None,
+        inflexible_device_sensors: list[int] | None = None,
+    ):
+        """Post schedule trigger with initial and target states of charge (soc)."""
+        message = {
+            "start": pd.Timestamp(
+                start
+            ).isoformat(),  # for example: 2021-10-13T00:00+02:00
+            "flex-model": {
+                "soc-unit": soc_unit,
+                "soc-at-start": soc_at_start,
+                "soc-targets": soc_targets,
+            },
+            "flex-context": {},
+        }
+
+        # Set optional flex context
+        if consumption_price_sensor is not None:
+            message["flex-context"][
+                "consumption-price-sensor"
+            ] = consumption_price_sensor
+        if production_price_sensor is not None:
+            message["flex-context"]["production-price-sensor"] = production_price_sensor
+        if inflexible_device_sensors is not None:
+            message["flex-context"][
+                "inflexible-device-sensors"
+            ] = inflexible_device_sensors
+
+        response, status = await self.request(
+            uri=f"sensors/{sensor_id}/schedules/trigger",
+            json=message,
+        )
+        check_for_status(status, 200)
+        print("Schedule triggered successfully.")
+
+    async def get_schedule(
+        self,
+        sensor_id: int,
+        schedule_id: str,
+        duration: str,
+    ):
+        """Get schedule with given ID."""
+        response, status = await self.request(
+            uri=f"sensors/{sensor_id}/schedules/{schedule_id}",
+            method="GET",
+            json={
+                "duration": pd.Timedelta(duration).isoformat(),  # for example: PT1H
+            },
+        )
+        check_for_status(status, 200)
+
+        return response, status
+
+    async def get_assets(self):
+        """Get all the assets available to the current user"""
+        response, status = await self.request(uri="assets", method="GET")
+        check_for_status(status, 200)
+        return response, status
```

Comment:
I think `client.get_assets` should return a list of `Asset` objects (where `Asset` is a simple data class with the attributes that are returned by our API). We could even derive that class dynamically, so in case our API starts returning additional attributes, they will automatically be picked up. Something like:
```
# in utils.py
from dataclasses import make_dataclass
from typing import Any

def obj_from_dict(t: str, d: dict) -> Any:
    cls = make_dataclass(t, list(zip(d.keys(), [type(v) for v in d.values()])))
    obj = cls(**d)
    return obj

# in get_assets
return obj_from_dict("Asset", response.json())
```
And perhaps we can rename the existing function to `client._get_assets`.

Reply:
Create an issue for this. Just return a dict for now

Notes for the issue: 
- will this allow for type hinting
- compare this approach with Pydantic

Reply:
I removed the return status for these functions. They are handled either before or unhelpful after this point in the flow. So now the response dict/list is returned.